scripts/src/main/resources/scripts/command/eclipse

```diff
@@ -23,19 +23,80 @@ source "$(dirname "${0}")"/../functions
+    doEcho "No plugins to install."
+    return
   fi
-  return "${result}"
+
+  doCheckPluginsInstalled "${plugins}"
+
+  # Install plugins and write debug info into temp file
+  local debug_info_file
+  debug_info_file=$(mktemp)
+  "${ECLIPSE_HOME}/eclipsec" -consoleLog -nosplash -vm "${ECLIPSE_JAVA_HOME}/bin/java" -application org.eclipse.equinox.p2.director -repository "${2}" -installIU "${1}" -vmargs "${ECLIPSE_VMARGS}" > "${debug_info_file}" 2>&1
+  exit_code=$?
+  # Write only lines with !MESSAGE into debug info
+  debug_info=$(grep -E "!MESSAGE" "${debug_info_file}")
+  
+
+  if [[ "${debug_info}" == *"Overall install request is satisfiable"* ]] && [ "${exit_code}" == 0 ]; then
+    rm "${debug_info_file}"
+    return 0
+  elif [[ "${debug_info}" == *"Unable to load repositories"* ]] || [[ "${debug_info}" == *"No repository found at"* ]]; then
+    doError "The given repository ${2} is not a correct update site."
+    rm "${debug_info_file}"
+    return 1
+  else
+    echo
+    doError "Installation of plugin '${plugin_name}' failed: \n${debug_info}"
+    rm "${debug_info_file}"
+    if [ -n "${plugin_dependencies}" ]
+    then
+      plugin_dependencies=$(echo -e "${plugin_dependencies}" | sort -u)
+      doWarning "Plugin '${plugin_name}' might be already installed, because of existing dependencies: ${plugin_dependencies} \nMake sure the plugin is not installed yet by checking the list of installed plugins in the Eclipse IDE. If there's an error and the issue persists, try to remove the plugin manually from the plugins folder or uninstall it with devon remove-plugin command."
+    fi
+    return ${exit_code}
+  fi
+}
+
+# Check if the plugins are already installed by looking at the eclipse/plugins folder
+# $1: plugin feature group ID
+# return: dependencies that are already installed
+function doCheckPluginsInstalled() {
+  local plugin_dir="${DEVON_IDE_HOME}/software/eclipse/plugins"
+
+  # Remove spaces from plugin variable
+  plugins="${1// /}"
+  # Iterate each plugin via comma seperation
+  IFS=',' read -ra plugins_arr <<< "${plugins}"
+  for plugin in "${plugins_arr[@]}"; do
+    # Remove feature from plugin variable
+    plugin=${plugin/\.feature*/}
+    # shellcheck disable=SC2010
+    plugin_dependencies=$(ls -1 "${plugin_dir}" | grep -o -E "${plugin}\..*\.jar")
+    # Remove last word after dot
+    while [[ "${plugin}" == *.* ]]; do
+      if [ "${plugin}" == "org.eclipse" ] || [ "${plugin}" == "org.apache" ] || [ "${plugin}" == "com" ] || [ "${plugin}" == "org" ] || [ "${plugin}" == "net" ]; then
+        break
+      fi
+      # shellcheck disable=SC2010
+      new_plugin_dependencies=$(ls -1 "${plugin_dir}" | grep -o -E "${plugin}\..*\.jar")
+      if [ -n "${new_plugin_dependencies}" ] 
+      then
+        plugin_dependencies="${plugin_dependencies}\n${new_plugin_dependencies}"
+      fi
+      plugin=${plugin%.*}
+    done
```

Comment:
So you implemented logic to detect if a plugin is present in Eclipse. In Bash. Wow!
However, I am concerned that we can do this in a reliable way.
Are you aware of the distinction between eclipse `plugins` and `features` folder?
Are you aware that a plugin can exist as *.jar file OR as a extracted folder and both variants are legal cases?
Do you know that you are relying on a naming convention of `plugin group ID` and `plugin artifact ID` that might be kept as best practices by most eclipse plugins but maybe not by all of them...

OK, if we use this to see if the plugin is already installed and if we find this applied we do not install again most probably the problematic case will lead to us not detecting that the plugin is already installed in an odd case so we keep the existing behavior to install it then anyway so maybe this is fine...

Reply:
Indeed, this way of plugin detection is not reliable at all. It can only point out to a potential problem if the problem is detected. And the problem might not always be because one of the given dependencies are already installed. Despite of this, I think it is good idea to point out to the user, that some of the required dependencies are installed already. At the end of the day, it only gives out a warning in the console and doesn't do any error correction itself, as I'd need a lot more knowledge on how the plugin/feature system in Eclipse IDE works to be able to correct the error.

Reply:
The suggestions make a lot of sense and I wish I would have known earlier, that a folder can also be seen as a plugin. In most cases however, I have observed, that when installing a plugin from the Eclipse IDE, new files are created in `features` folder as well as in `plugins` folder. If you don't want to iterate the folders themselves, the same logic from above can also be used with the `artifacts.xml` file, which contains all of the plugins id's.